Raven.Abstractions/Smuggler/SmugglerDatabaseApiBase.cs

```diff
@@ -128,17 +128,17 @@ protected bool IgnoreErrorsAndContinue
                         await ExportIndexes(exportOptions.From, jsonWriter);
                     }
                     jsonWriter.WriteEndArray();
-
-                    // used to synchronize max returned values for put/delete operations
-                    var maxEtags = Operations.FetchCurrentMaxEtags();
-
+                    
+                    var lastDocDeleteEtag = Operations.FetchLastDocDeleteEtag();
                     jsonWriter.WritePropertyName("Docs");
                     jsonWriter.WriteStartArray();
                     if (Options.OperateOnTypes.HasFlag(ItemType.Documents))
                     {
                         try
                         {
-                            result.LastDocsEtag = await ExportDocuments(exportOptions.From, jsonWriter, result.LastDocsEtag, maxEtags.LastDocsEtag);
+                            result.LastDocsEtag = await ExportDocuments(exportOptions.From, jsonWriter, 
```

Comment:
@grisha-kotler & @ppekrol

I suggest you do a skype call to review this code.

*Hibernating Rhinos Ltd  *

Oren Eini* l CEO l *Mobile: + 972-52-548-6969

Office: +972-4-622-7811 *l *Fax: +972-153-4-622-7811



On Thu, Jan 14, 2016 at 4:38 PM, grisha-kotler <notifications@github.com>
wrote:

> In Raven.Abstractions/Smuggler/SmugglerDatabaseApiBase.cs
> <https://github.com/ayende/ravendb/pull/2963#discussion_r49732577>:
>
> >                      jsonWriter.WritePropertyName("Docs");
> >                      jsonWriter.WriteStartArray();
> >                      if (Options.OperateOnTypes.HasFlag(ItemType.Documents))
> >                      {
> >                          try
> >                          {
> > -                            result.LastDocsEtag = await ExportDocuments(exportOptions.From, jsonWriter, result.LastDocsEtag, maxEtags.LastDocsEtag);
> > +                            result.LastDocsEtag = await ExportDocuments(exportOptions.From, jsonWriter,
>
> using maxEtags will mean that we'll have missing documents.
> see RaveDB-4085
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ayende/ravendb/pull/2963/files#r49732577>.
>